File: 258. Add Digits/solution.cpp
/*
0 ms
5.9 MB

TC : O(N*N)
SC : O(N)
*/

/**
 * @author : SahilK-027
 * @brief : Recursive solution
*/

class Solution {
private:
    int solve(int n){
        if(n / 10 <= 0){
            return n;
        }
        int sum = 0;
        while(n > 0){
            sum += n % 10;
            n /= 10;
        }
        return solve(sum);
    }
public:
    int addDigits(int num) {
        return solve(num);
    }
};

File: 258. Add Digits/solution.js
/*
76 ms
43.6 MB

TC : O(1)
SC : O(1)
*/

/**
 * @author : SahilK-027
 * @brief : Math
*/

/**
 * @param {number} num
 * @return {number}
 */
var addDigits = function(num) {
    if(num==0) return 0;
    else if(num%9==0) return 9;
    else return num%9;
};